### ISSUE TITLE
Something weird with slot co-op

### ISSUE THREAD

**Issue by ScipioWright** [DaftBrit/NoitaArchipelago#130]
Trap items in the shop don't disappear when your co-op partner picks them up (the other items disappear fine).

Also something weird with the orbs not disappearing properly in slot co-op, or occasionally not giving the location check. Very unclear on this one. Happens both in the shop and on pedestals.

**Comment by ScipioWright** [comment]
It's possible the orb issue is because of ap_orb_init_randomizer_spawned being called when it shouldn't be. Like what was happening with the shop items before, it may be running this script multiple times (or changing the xml multiple time), and something in that could be removing the part where it sends the location ID or something, idk.

**Comment by heinermann** [comment]
You did remove a piece of code I thought you shouldn't have, which was the room update that told you which location got checked which removed stuff from the pool. I was gonna add it back in at some point while testing collect.

**Comment by ScipioWright** [comment]
Oh, I removed that because I didn't think we were using it for anything anymore. Slot co-op still works with other items just fine, just those couple things were causing issues.

Though also, something I just remembered: as part of the test, I went way down to each of the shops and picked up the item where the spell refresh would be, as well as some of the shop items, while my friend waited in the Mines. Then, he went down to several of the shops and found that the items replacing the spell refreshes were still there (and the spell refresh was not there), and that the items I bought from the shop were still there for him.

So, I think the issue is that when we remove the item, we're not also removing the item's listing from whatever global we're using to see which items to spawn or not. I guess I kind of figured that would work on its own, but it's an easy enough fix.
